plugins/filter/merge_data.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2022     , DEMAREST Maxime <maxime@indelog.fr>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+name: merge_data
+short_description: Compare one data structure provided in I(_input) with an
+                   other provided in I(expected) and ensure that these data are
+                   presents or absents in the result depending the I(state).
+version_added: 5.2.0
+author: DEMAREST Maxime (@indelog)
+description:
+  - Compare data structure provided in I(_input) with data structure provided
```

Comment:
Hmm, I don't think "compare" is the correct verb here. The filter returns a modified copy of the original data. To compute the result it does compare something, but more importantly is that it returns a modified copy.